app/app.go

```diff
@@ -1488,6 +1490,46 @@ func (d *Deployments) GetDevicesListForDeployment(ctx context.Context,
 	return statuses, totalCount, nil
 }
 
+func (d *Deployments) GetDeviceDeploymentListForDevice(ctx context.Context,
+	query store.ListQueryDeviceDeployments) ([]model.DeviceDeploymentListItem, int, error) {
+	deviceDeployments, totalCount, err := d.db.GetDeviceDeploymentsForDevice(ctx, query)
+	if err != nil {
+		return nil, -1, errors.Wrap(err, "retrieving the list of deployment statuses")
+	}
+
+	deploymentIDs := make([]string, len(deviceDeployments))
+	for i, deviceDeployment := range deviceDeployments {
+		deploymentIDs[i] = deviceDeployment.DeploymentId
+	}
+
+	deployments, _, err := d.db.Find(ctx, model.Query{
+		IDs:   deploymentIDs,
+		Limit: len(deviceDeployments),
+	})
+	if err != nil {
+		return nil, -1, errors.Wrap(err, "retrieving the list of deployments")
+	}
+
+	deploymentsMap := make(map[string]*model.Deployment, len(deployments))
+	for _, deployment := range deployments {
+		deploymentsMap[deployment.Id] = deployment
+	}
+
+	res := make([]model.DeviceDeploymentListItem, 0, len(deviceDeployments))
+	for _, deviceDeployment := range deviceDeployments {
+		if deployment, ok := deploymentsMap[deviceDeployment.DeploymentId]; ok {
+			device := deviceDeployment
+			res = append(res, model.DeviceDeploymentListItem{
+				Id:         deviceDeployment.Id,
+				Deployment: deployment,
+				Device:     &device,
+			})
+		}
```

Comment:
Although this shouldn't happen, we should provide the incomplete information we have to not mess up the pagination.
```suggestion
		} else {
			device := deviceDeployment
			res = append(res, model.DeviceDeploymentListItem{
				Id:         deviceDeployment.Id,
				Device:     &device,
			})
		}
```